src/classes/DiscordBot.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-assignment */
+
+import { Client, Intents, Message } from 'discord.js';
+import log from '../lib/logger';
+import Options from './Options';
+import Bot from './Bot';
+import SteamID from 'steamid';
+//import Commands from './Commands';
+
+export default class DiscordBot {
+    readonly client: Client;
+
+    //readonly commands: Commands;
+
+    constructor(private options: Options, private bot: Bot) {
+        this.client = new Client({
+            intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES]
+        });
+        //this.client.login(process.env.TOKEN);
+        this.client.login(this.options.discordApiKey);
+        //this.commands = new Commands();
+    }
+
+    public start(): void {
+        this.client.on('ready', this.ClientReady.bind(this));
+        this.client.on('messageCreate', async message => this.handleMessage(message));
+    }
+
+    public stop(): void {
+        //this.commands.stop();
+        log.info('Logging out from Discord...');
+        this.client.destroy();
+    }
+
+    public async handleMessage(message: Message): Promise<void> {
+        if (message.author === this.client.user) {
+            return; // don't talk to myself
+        }
+
+        log.info(`Got new message ${String(message.content)} from ${String(message.author.id)}`);
+        if (message.author.id !== this.options.discordAdmin) {
+            return; // obey only admin
+        }
+
+        // TODO: don't handle messages until bot is fully up
+
+        const ans = `I see your "${String(message.content)}"! Answered in Steam for now.`;
+        await message.reply(ans);
+
+        const adminID = new SteamID(this.options.steamOfDiscordAdmin);
+        await this.bot.handler.onMessage(adminID, message.content); // TODO: redirect answer to Discord
```

Comment:
It may be worthwhile to instead use slash commands, as Discord seems to be against the idea of reading message contents. Slash commands are also much more versatile and has "richer" interactions

Reply:
This will require far more work with code. For now bot commands described in `Commands.ts` and in `Help.ts`. Showing it like slash commands in Discord will require either 3rd place where commands specified (to send the full list to Discord) or big rework for something like `Command` class. I consider first of this variants as bad, and second - as long.

For now I have 2 ideas about it. First - just leave `!` as prefix, which will lead to the least inconvenience for the user (because at this moment all commands go with `!` prefix). Second - to make an option to change prefix in order to manage multiple bots with one Discord bot (I'm not sure if Discord allows to connect more than 1 app to Discord bot account, but if yes - it'll be far more comfortably for the user who runs more than one tf2autobot instance).

For now Discord bot doesn't require any approval and doesn't get any restrictions for reading messages if the bot connected to less than 100 (some says 75) servers. I'm sure this is our case.